src/main/java/datawave/query/util/AllFieldMetadataHelper.java

```diff
@@ -1061,350 +1059,307 @@ public Set<String> loadDatatypes() throws TableNotFoundException {
     /**
      * Fetches results from {@link #metadataTableName} and calculates the set of field index holes that exists for all indexed entries. The map consists of
      * field names to datatypes to field index holes.
-     * 
+     *
+     * @param minThreshold
+     *            the minimum percentage threshold required for an index row to be considered NOT a hole on a particular date, this should be a value in the
+     *            range 0.0 to 1.0
      * @return a map of field names and datatype pairs to field index holes
      */
     @Cacheable(value = "getFieldIndexHoles", key = "{#root.target.auths,#root.target.metadataTableName}", cacheManager = "metadataHelperCacheManager")
-    public Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles() throws TableNotFoundException, CharacterCodingException {
-        return getFieldIndexHoles(ColumnFamilyConstants.COLF_I);
+    public Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(double minThreshold) throws TableNotFoundException, IOException {
+        return getFieldIndexHoles(ColumnFamilyConstants.COLF_I, minThreshold);
     }
     
     /**
      * Fetches results from {@link #metadataTableName} and calculates the set of field index holes that exists for all reversed indexed entries. The map
      * consists of field names to datatypes to field index holes.
-     * 
+     *
+     * @param minThreshold
+     *            the minimum percentage threshold required for an index row to be considered NOT a hole on a particular date, this should be a value in the
+     *            range 0.0 to 1.0
      * @return a map of field names and datatype pairs to field index holes
      */
     @Cacheable(value = "getReversedFieldIndexHoles", key = "{#root.target.auths,#root.target.metadataTableName}", cacheManager = "metadataHelperCacheManager")
-    public Map<String,Map<String,FieldIndexHole>> getReversedFieldIndexHoles() throws TableNotFoundException, CharacterCodingException {
-        return getFieldIndexHoles(ColumnFamilyConstants.COLF_RI);
+    public Map<String,Map<String,FieldIndexHole>> getReversedFieldIndexHoles(double minThreshold) throws TableNotFoundException, IOException {
+        return getFieldIndexHoles(ColumnFamilyConstants.COLF_RI, minThreshold);
     }
     
-    /**
-     * Supplies field index hole for {@link #getFieldIndexHoles()} and {@link #getReversedFieldIndexHoles()}.
-     */
-    private Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(Text indexColumnFamily) throws TableNotFoundException {
+    private Map<String,Map<String,FieldIndexHole>> getFieldIndexHoles(Text targetColumnFamily, double minThreshold) throws TableNotFoundException, IOException {
```

Comment:
should we pass in a datatype filter?

Reply:
We can add filters for both field names and datatypes easily enough. Are these results cached on a per-query basis or for longer than that?